homeassistant/components/qwikswitch.py

```diff
@@ -0,0 +1,180 @@
+"""
+Support for Qwikswitch lights and switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/qwikswitch
+"""
+
+import logging
+
+REQUIREMENTS = ['https://github.com/kellerza/pyqwikswitch/archive/v0.1.zip'
+                '#pyqwikswitch==0.1']
+DEPENDENCIES = []
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'qwikswitch'
+DISCOVER_LIGHTS = 'qwikswitch.light'
+DISCOVER_SWITCHES = 'qwikswitch.switch'
+
+ADD_DEVICES = None
+
+
+class QSToggleEntity(object):
+    # pylint: disable=line-too-long
+    """Representation of a Qwikswitch Entiry.
+
+    Implement base QS methods. Modeled around HA ToggleEntity[1] & should only
+    be used in a class that extends both QSToggleEntity *and* ToggleEntity.
+
+    Implemented:
+     - QSLight extends QSToggleEntity and Light[2] (ToggleEntity[1])
+     - QSSwitch extends QSToggleEntity and SwitchDevice[3] (ToggleEntity[1])
+
+    [1] https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity.py  # NOQA
```

Comment:
There is no need to paste full github urls in the code base, just absolute paths based on project root should be enough.